packages/adapter-node/src/handler.js

```diff
@@ -7,17 +7,32 @@ import { getRequest, setResponse } from '@sveltejs/kit/node';
 import { Server } from 'SERVER';
 import { manifest } from 'MANIFEST';
 
-/* global ORIGIN, ADDRESS_HEADER, PROTOCOL_HEADER, HOST_HEADER, X_FORWARDED_FOR_INDEX */
+/* global ORIGIN, ADDRESS_HEADER, PROTOCOL_HEADER, HOST_HEADER */
 
 const server = new Server(manifest);
 const origin = ORIGIN;
 
+const xff_depth = get_xff_depth();
 const address_header = ADDRESS_HEADER && (process.env[ADDRESS_HEADER] || '').toLowerCase();
 const protocol_header = PROTOCOL_HEADER && process.env[PROTOCOL_HEADER];
 const host_header = (HOST_HEADER && process.env[HOST_HEADER]) || 'host';
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
+function get_xff_depth() {
+	const value = process.env['XFF_DEPTH'];
+	let xff_depth;
+	try {
+		xff_depth = value ? parseInt(value) : 1;
```

Comment:
this shouldn't be in a try-catch. `parseInt` will never throw, it'll just return `NaN`

Reply:
opened https://github.com/sveltejs/kit/pull/4357 as it was easier than piecemeal suggestions for a replacement for `get_xff_depth`